test/lit/passes/merge-similar-functions_types.wast

```diff
@@ -4,24 +4,37 @@
 
 ;; Calls to functions $2 and $3 are the only differences between the contents
 ;; of $0 and $1, so we want to merge them and pass ref.funcs of $2 and $3.
-;; However, their nominal types differ, so in nominal typing we cannot do so.
+;; However, their types differ, so we cannot do so.
 (module
- ;; CHECK:      (type $type$0 (sub (func)))
- (type $type$0 (sub (func)))
- (type $type$1 (sub (func)))
- (type $type$2 (sub (func)))
- (type $type$3 (sub (func (param f32) (result f32))))
- (type $type$4 (sub (func (param f64) (result f64))))
- ;; CHECK:      (type $1 (func (param (ref $type$0))))
+ (rec
+   ;; CHECK:      (type $0 (func))
 
- ;; CHECK:      (elem declare func $2 $3)
-
- ;; CHECK:      (func $0 (type $type$0)
- ;; CHECK-NEXT:  (return_call $byn$mgfn-shared$0
- ;; CHECK-NEXT:   (ref.func $2)
- ;; CHECK-NEXT:  )
+   ;; CHECK:      (rec
+   ;; CHECK-NEXT:  (type $type$1 (sub (func)))
+   (type $type$1 (sub (func)))
+   ;; CHECK:       (type $type$2 (sub (func)))
+   (type $type$2 (sub (func)))
+ )
+ ;; CHECK:      (func $0 (type $0)
+ ;; CHECK-NEXT:  (nop)
+ ;; CHECK-NEXT:  (nop)
+ ;; CHECK-NEXT:  (nop)
+ ;; CHECK-NEXT:  (nop)
+ ;; CHECK-NEXT:  (nop)
+ ;; CHECK-NEXT:  (nop)
+ ;; CHECK-NEXT:  (nop)
+ ;; CHECK-NEXT:  (nop)
+ ;; CHECK-NEXT:  (call $2)
+ ;; CHECK-NEXT:  (nop)
+ ;; CHECK-NEXT:  (nop)
+ ;; CHECK-NEXT:  (nop)
+ ;; CHECK-NEXT:  (nop)
+ ;; CHECK-NEXT:  (nop)
+ ;; CHECK-NEXT:  (nop)
+ ;; CHECK-NEXT:  (nop)
+ ;; CHECK-NEXT:  (nop)
```

Comment:
Why did this change?

Reply:
The test was not previously testing what it claimed to test, which was the case where `$types$1` is not the same as `$type$2`. After fixing the types to match the intent of the test, the optimization no longer applies.